fl4health/feature_alignment/tab_features_preprocessor.py

```diff
@@ -0,0 +1,147 @@
+from typing import Dict, List, Tuple
+
+import pandas as pd
+from flwr.common.typing import NDArray, Scalar
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import MinMaxScaler, OneHotEncoder, OrdinalEncoder
+
+from fl4health.feature_alignment.string_columns_transformer import TextColumnTransformer
+from fl4health.feature_alignment.tab_features_info_encoder import TabularFeaturesInfoEncoder
+from fl4health.feature_alignment.tabular_feature import MetaData, TabularFeature
+from fl4health.feature_alignment.tabular_type import TabularType
+
+
+class TabularFeaturesPreprocessor:
+    """
+    TabularFeaturesPreprocessor is responsible for constructing
+    the appropriate column transformers based on the information
+    encoded in tab_feature_encoder. These transformers will
+    then be applied to a pandas dataframe.
+
+    Each tabular feature, which corresponds to a column
+    in the pandas dataframe, has its own column transformer. A default
+    transformer is initialized for each feature based on its data type,
+    but the user may also manually specify a transformer for this
+    feature.
+
+    Parameters
+    ----------
+    tab_feature_encoder: TabularFeaturesInfoEncoder
+        Encodes the information necessary for constructing the column transformers.
+    """
+
+    def __init__(self, tab_feature_encoder: TabularFeaturesInfoEncoder) -> None:
+        self.features_to_pipelines: Dict[str, Pipeline] = {}
+        self.targets_to_pipelines: Dict[str, Pipeline] = {}
+
+        self.tabular_features = tab_feature_encoder.get_tabular_features()
+        self.tabular_targets = tab_feature_encoder.get_tabular_targets()
+
+        self.feature_columns = tab_feature_encoder.get_feature_columns()
+        self.target_columns = tab_feature_encoder.get_target_columns()
+
+        self.initialize_default_pipelines(self.tabular_features, self.features_to_pipelines, one_hot=True)
+        self.initialize_default_pipelines(self.tabular_targets, self.targets_to_pipelines, one_hot=False)
+
+        self.data_column_transformer = self.return_column_transformer(self.features_to_pipelines)
+        self.target_column_transformer = self.return_column_transformer(self.targets_to_pipelines)
+
+    def get_default_numeric_pipeline(self) -> Pipeline:
+        return Pipeline(steps=[("imputer", SimpleImputer(strategy="mean")), ("scaler", MinMaxScaler())])
+
+    def get_default_binary_pipeline(self) -> Pipeline:
+        return Pipeline(steps=[("imputer", SimpleImputer(strategy="most_frequent")), ("encoder", OrdinalEncoder())])
+
+    def get_one_hot_pipeline(self, categories: MetaData) -> Pipeline:
```

Comment:
Just a suggestion, but for this function, and `get_ordinal_pipeline` below, maybe we follow the same naming convention as above and prefix them as `get_default_...` to be consistent?